source/jormungandr/jormungandr/street_network/here.py

```diff
@@ -338,41 +310,35 @@ def _convert_direction(direction):
             else:
                 return 0
 
-        for maneuver in leg.get('maneuver', []):
+        # instruction
+        i = 0
+        for maneuver in here_section.get('actions', []):
             path_item = section.street_network.path_items.add()
-            path_item.id = int(maneuver.get('id', -1).replace('M', ''))
+            path_item.id = i
             path_item.length = maneuver.get('length', 0)
-            path_item.duration = maneuver.get('travelTime', 0)
+            path_item.duration = maneuver.get('duration', 0)
             path_item.instruction = maneuver.get('instruction', '')
-            path_item.name = maneuver.get('roadName', '')
+            path_item.name = ''
             path_item.direction = _convert_direction(maneuver.get('direction', ''))
-            path_item.instruction_start_coordinate.lat = float(
-                maneuver.get('position', 0.0).get('latitude', 0.0)
-            )
-            path_item.instruction_start_coordinate.lon = float(
-                maneuver.get('position', 0.0).get('longitude', 0.0)
-            )
-
-        shape = route.get('shape', [])
-        for sh in shape:
-            coord = section.street_network.coordinates.add()
-            lat, lon = sh.split(',')
-            coord.lon = float(lon)
-            coord.lat = float(lat)
+            offset = maneuver.get('offset', 0)
+            if offset < shape_len:
+                path_item.instruction_start_coordinate.lat = float(shape[offset][0])
+                path_item.instruction_start_coordinate.lon = float(shape[offset][1])
+            i = i + 1
```

Comment:
to remove

Reply:
Nope, we need it to catch the good coord with the available offset